woocommerce/woocommerce-gateway-stripe
# Fix deprecated messages on PHP 8.2 from using FILTER_SANITIZE_STRING

## Changes proposed in this Pull Request:







While testing another PR (https://github.com/woocommerce/woocommerce-gateway-stripe/issues/2837) I noticed a bunch of deprecated warnings in my error logs while trying to purchase a product using Google Pay and Apple Pay:

```
[22-May-2024 05:10:49 UTC] PHP Deprecated:  Constant FILTER_SANITIZE_STRING is deprecated in /Users/matt/local/woo/wp-content/plugins/woocommerce-gateway-stripe/includes/payment-methods/class-wc-stripe-payment-request.php on line 1175
[22-May-2024 05:10:49 UTC] PHP Deprecated:  Constant FILTER_SANITIZE_STRING is deprecated in /Users/matt/local/woo/wp-content/plugins/woocommerce-gateway-stripe/includes/payment-methods/class-wc-stripe-payment-request.php on line 1176
[22-May-2024 05:10:49 UTC] PHP Deprecated:  Constant FILTER_SANITIZE_STRING is deprecated in /Users/matt/local/woo/wp-content/plugins/woocommerce-gateway-stripe/includes/payment-methods/class-wc-stripe-payment-request.php on line 1177
[22-May-2024 05:10:49 UTC] PHP Deprecated:  Constant FILTER_SANITIZE_STRING is deprecated in /Users/matt/local/woo/wp-content/plugins/woocommerce-gateway-stripe/includes/payment-methods/class-wc-stripe-payment-request.php on line 1178
[22-May-2024 05:10:49 UTC] PHP Deprecated:  Constant FILTER_SANITIZE_STRING is deprecated in /Users/matt/local/woo/wp-content/plugins/woocommerce-gateway-stripe/includes/payment-methods/class-wc-stripe-payment-request.php on line 1179
[22-May-2024 05:10:49 UTC] PHP Deprecated:  Constant FILTER_SANITIZE_STRING is deprecated in /Users/matt/local/woo/wp-content/plugins/woocommerce-gateway-stripe/includes/payment-methods/class-wc-stripe-payment-request.php on line 1180
[22-May-2024 05:10:49 UTC] PHP Deprecated:  Constant FILTER_SANITIZE_STRING is deprecated in /Users/matt/local/woo/wp-content/plugins/woocommerce-gateway-stripe/includes/payment-methods/class-wc-stripe-payment-request.php on line 1183
```

Looks like the [`FILTER_SANITIZE_STRING` constant](https://www.php.net/manual/en/filter.constants.php#constant.filter-sanitize-string) has been deprecated since PHP 8.1 and so this PR updates our code to use `FILTER_SANITIZE_SPECIAL_CHARS` instead which is essentially the same thing.


## Testing instructions



1. Switch your dev site over to PHP 8.2+
2. Enable Apple Pay/Google Pay
3. While on `develop` add a product to your cart and attempt to purchase it using Google Pay
4. When the Google Pay window appears, check you debug log and confirm the deprecated messages
5. Checkout this branch and confirm no deprecation messages
6. These changes impact fetching shipping options so confirm the shipping based on your address are correct



---


**Post merge**


Fixes #3056
